File: app/src/main/java/com/fieldservice/app/data/mock/MockTicketData.java
package com.fieldservice.app.data.mock;

import com.fieldservice.app.domain.model.Priority;
import com.fieldservice.app.domain.model.Ticket;
import com.fieldservice.app.domain.model.TicketStatus;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

/**
 * MOCK: chamados de exemplo usados enquanto não existe uma API real.
 * Serve apenas para desenvolvimento/demonstração da primeira versão do app.
 */
public final class MockTicketData {

    private MockTicketData() {
    }

    private static Date daysAgo(int days) {
        Calendar calendar = Calendar.getInstance();
        calendar.add(Calendar.DAY_OF_YEAR, -days);
        return calendar.getTime();
    }

    public static List<Ticket> initialTickets() {
        List<Ticket> tickets = new ArrayList<>();

        tickets.add(new Ticket(
                "1028",
                "#1028",
                "Servidor indisponível",
                "Servidor não inicializa após queda de energia.",
                "Empresa XYZ",
                "Brasília - DF",
                Priority.HIGH,
                TicketStatus.ASSIGNED,
                daysAgo(0)
        ));
        tickets.add(new Ticket(
                "1029",
                "#1029",
                "Falha de conexão de rede",
                "Estações de trabalho perdem conexão com a rede local intermitentemente.",
                "Empresa ABC",
                "São Paulo - SP",
                Priority.MEDIUM,
                TicketStatus.ACCEPTED,
                daysAgo(1)
        ));
        tickets.add(new Ticket(
                "1030",
                "#1030",
                "Impressora indisponível",
                "Impressora do setor financeiro não liga.",
                "Empresa Delta",
                "Belo Horizonte - MG",
                Priority.LOW,
                TicketStatus.COMPLETED,
                daysAgo(3)
        ));
        tickets.add(new Ticket(
                "1031",
                "#1031",
                "Sistema de backup falhando",
                "Rotina de backup noturno falhou nas últimas três execuções.",
                "Empresa Omega",
                "Curitiba - PR",
                Priority.CRITICAL,
                TicketStatus.OPEN,
                daysAgo(0)
        ));
        tickets.add(new Ticket(
                "1032",
                "#1032",
                "Instalação de novo equipamento",
                "Instalação e configuração de nova estação de trabalho.",
                "Empresa Prime",
                "Porto Alegre - RS",
                Priority.MEDIUM,
                TicketStatus.IN_PROGRESS,
                daysAgo(2)
        ));
        tickets.add(new Ticket(
                "1033",
                "#1033",
                "Manutenção preventiva",
                "Manutenção preventiva trimestral do parque de máquinas.",
                "Empresa Nova",
                "Recife - PE",
                Priority.LOW,
                TicketStatus.COMPLETED,
                daysAgo(5)
        ));
        tickets.add(new Ticket(
                "1034",
                "#1034",
                "Rede Wi-Fi instável",
                "Sinal de Wi-Fi cai constantemente no segundo andar do escritório.",
                "Empresa Zenith",
                "Brasília - DF",
                Priority.HIGH,
                TicketStatus.TRAVELING,
                daysAgo(0)
        ));

        return tickets;
    }
}

File: legacy-kotlin/app/src/main/java/com/fieldservice/app/data/mock/MockTicketData.kt
package com.fieldservice.app.data.mock

import com.fieldservice.app.domain.model.Priority
import com.fieldservice.app.domain.model.Ticket
import com.fieldservice.app.domain.model.TicketStatus
import java.util.Calendar
import java.util.Date

/**
 * MOCK: chamados de exemplo usados enquanto não existe uma API real.
 * Serve apenas para desenvolvimento/demonstração da primeira versão do app.
 */
object MockTicketData {

    private fun daysAgo(days: Int): Date {
        val calendar = Calendar.getInstance()
        calendar.add(Calendar.DAY_OF_YEAR, -days)
        return calendar.time
    }

    fun initialTickets(): List<Ticket> = listOf(
        Ticket(
            id = "1028",
            number = "#1028",
            title = "Servidor indisponível",
            description = "Servidor não inicializa após queda de energia.",
            customerName = "Empresa XYZ",
            address = "Brasília - DF",
            priority = Priority.HIGH,
            status = TicketStatus.ASSIGNED,
            createdAt = daysAgo(0)
        ),
        Ticket(
            id = "1029",
            number = "#1029",
            title = "Falha de conexão de rede",
            description = "Estações de trabalho perdem conexão com a rede local intermitentemente.",
            customerName = "Empresa ABC",
            address = "São Paulo - SP",
            priority = Priority.MEDIUM,
            status = TicketStatus.ACCEPTED,
            createdAt = daysAgo(1)
        ),
        Ticket(
            id = "1030",
            number = "#1030",
            title = "Impressora indisponível",
            description = "Impressora do setor financeiro não liga.",
            customerName = "Empresa Delta",
            address = "Belo Horizonte - MG",
            priority = Priority.LOW,
            status = TicketStatus.COMPLETED,
            createdAt = daysAgo(3)
        ),
        Ticket(
            id = "1031",
            number = "#1031",
            title = "Sistema de backup falhando",
            description = "Rotina de backup noturno falhou nas últimas três execuções.",
            customerName = "Empresa Omega",
            address = "Curitiba - PR",
            priority = Priority.CRITICAL,
            status = TicketStatus.OPEN,
            createdAt = daysAgo(0)
        ),
        Ticket(
            id = "1032",
            number = "#1032",
            title = "Instalação de novo equipamento",
            description = "Instalação e configuração de nova estação de trabalho.",
            customerName = "Empresa Prime",
            address = "Porto Alegre - RS",
            priority = Priority.MEDIUM,
            status = TicketStatus.IN_PROGRESS,
            createdAt = daysAgo(2)
        ),
        Ticket(
            id = "1033",
            number = "#1033",
            title = "Manutenção preventiva",
            description = "Manutenção preventiva trimestral do parque de máquinas.",
            customerName = "Empresa Nova",
            address = "Recife - PE",
            priority = Priority.LOW,
            status = TicketStatus.COMPLETED,
            createdAt = daysAgo(5)
        ),
        Ticket(
            id = "1034",
            number = "#1034",
            title = "Rede Wi-Fi instável",
            description = "Sinal de Wi-Fi cai constantemente no segundo andar do escritório.",
            customerName = "Empresa Zenith",
            address = "Brasília - DF",
            priority = Priority.HIGH,
            status = TicketStatus.TRAVELING,
            createdAt = daysAgo(0)
        )
    )
}
